src/prefect/runner/submit.py

```diff
@@ -94,62 +120,85 @@ async def _submit_flow_to_runner(
-) -> List[uuid.UUID]:
-    """
-    Run multiple callables in the background via the runner webserver, respecting the run limit.
-
-    Args:
-        prefect_callable: the callable to run (flows or tasks)
-        parameters_list: a list of dictionaries, each representing keyword arguments to pass to the callable.
+        parameters: keyword arguments to pass to the callable. May be a list of dictionaries where
+            each dictionary represents a discrete invocation of the callable
+        retry_failed_submissions: Whether to retry failed submissions to the runner webserver.
     """
+    if not PREFECT_EXPERIMENTAL_ENABLE_EXTRA_RUNNER_ENDPOINTS.value():
+        raise ValueError(
+            "The `submit_to_runner` utility requires the `Runner` webserver to be"
+            " built with extra endpoints enabled. To enable this, set the"
+            " `PREFECT_EXPERIMENTAL_ENABLE_EXTRA_RUNNER_ENDPOINTS` setting to `True`."
+        )
 
-    n_current_runs = await get_current_run_count()
-    available_slots = max(0, PREFECT_RUNNER_PROCESS_LIMIT.value() - n_current_runs)
+    if isinstance(parameters, dict):
+        parameters = [parameters]
 
     submitted_run_ids = []
-    for parameters in parameters_list[:available_slots]:
-        flow_run_id = await submit_to_runner(prefect_callable, parameters)
-        if inspect.isawaitable(flow_run_id):
-            flow_run_id = await flow_run_id
-        submitted_run_ids.append(flow_run_id)
+    unsubmitted_parameters = []
+
+    for p in parameters:
+        try:
+            flow_run_id = await _submit_flow_to_runner(
+                prefect_callable, p, retry_failed_submissions
+            )
+            if inspect.isawaitable(flow_run_id):
+                flow_run_id = await flow_run_id
+            submitted_run_ids.append(flow_run_id)
+        except (httpx.ConnectError, httpx.HTTPStatusError) as exc:
+            if PREFECT_RUNNER_SERVER_ENABLE_BLOCKING_FAILOVER.value():
+                logger.warning(
+                    "The `submit_to_runner` utility failed to connect to the `Runner`"
+                    " webserver, but blocking failover is enabled. The utility will"
+                    " block until the flow run completes. You can disable this behavior"
+                    " by configuring the"
+                    " `PREFECT_RUNNER_SERVER_ENABLE_BLOCKING_FAILOVER` setting to"
+                    " `False`."
+                )
+                unsubmitted_parameters.append(p)
+            else:
+                raise RuntimeError(
+                    "The `submit_to_runner` utility failed to connect to the `Runner`"
+                    " webserver. To enable blocking failover, set the"
+                    " `PREFECT_RUNNER_SERVER_ENABLE_BLOCKING_FAILOVER` setting to"
+                    " `True`."
+                ) from exc
+
+    if unsubmitted_parameters:
+        await asyncio.gather(
```

Comment:
What do you think of collecting these IDs and merging them into the `submitted_run_ids` list? I think that would be a more truthful account of the flows that were executed.